pkg/controller/controller_test.go

```diff
@@ -572,21 +583,75 @@ func TestGetSecretReference(t *testing.T) {
 			expectRef:    nil,
 			expectErr:    true,
 		},
+		"simple - PVC name annotations not supported for Provision and Delete": {
+			secretParams: provisionerSecretParams,
+			params: map[string]string{
+				prefixedProvisionerSecretNameKey: "static-${pv.name}-${pvc.namespace}-${pvc.name}-${pvc.annotations['akey']}",
+			},
+			pvc: &v1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "name",
+					Namespace: "ns",
+				},
+			},
+			expectErr: true,
+		},
 		"template - valid": {
 			secretParams: nodePublishSecretParams,
 			params: map[string]string{
-				nodePublishSecretNameKey:      "static-${pv.name}-${pvc.namespace}-${pvc.name}-${pvc.annotations['akey']}",
+				nodePublishSecretNameKey:      "static-${pv.name}-${pvc.namespace}-${pvc.name}",
```

Comment:
annotations are allowed for nodepublish right?

Reply:
Yeah, they are. Just made this into two unit tests - one for nodePublish secrets and one for provisioner secrets